pages/en/zkapps/simple-anonymous-message-board-tutorial.mdx

```diff
@@ -0,0 +1,345 @@
+```
+
+This command creates a directory containing a new project template, fully set up
+& ready for local development.
+
+- See the included [README](templates/project-ts/README.md) for usage instructions.
+  All usual commands will be available: `npm run build`, `npm run test`,
+  `npm run coverage`, etc.
+- A Git repo will be initialized in the project directory automatically. For
+  consistency, we use `main` as the default Git branch, by convention.
+- A [Github Actions CI workflow](templates/project-ts/.github/workflows/ci.yml) is
+  also included. If you push your project to Github, Github Actions will run
+  your tests (named as `*.test.js`) automatically, whenever you push a commit or
+  open a pull request.
+- Code style consistency (via Prettier) and linting (via ES Lint) are
+  automatically enforced using Git pre-commit hooks. This requires no
+  configuration and occurs automatically when you commit to Git--e.g. `git commit -m 'feat: add awesome feature'`.
+- To skip all checks in the Git pre-commit hook (not recommended), you can pass
+  the `-n` flag to Git--e.g. `git commit -m 'a bad commit' -n`. But we'd
+  recommend avoiding this and resolving any errors which exist in your project
+  until the pre-commit hook passes.
+
+### Scaffolding
+
+Now that your project is set up, you can open it in your IDE or `cd zk-message` if you work from the command line.
+
+There should be an example smart contract in `./src` called `Add.ts` and tests for it in `Add.test.ts`. These are just example code, and you can delete them if they bother you.
+
+Let’s create a new contract by running:
+
+```sh
+zk file Message
+```
+
+Now open it (`Message.ts`) up and paste in the following:
+
+```ts
+import {
+  Field,
+  SmartContract,
+  state,
+  State,
+  method,
+  DeployArgs,
+  Permissions,
+  PrivateKey,
+  PublicKey,
+  isReady,
+  Poseidon,
+  Encoding,
+} from "snarkyjs";
+
+export { isReady, Field, Encoding };
+
+// Wait till our SnarkyJS instance is ready
+await isReady;
+
+// These private keys are exported so that experimenting with the contract is
+// easy. Three of them (the Bobs) are used when the contract is deployed to
+// generate the public keys that are allowed to post new messages. Jack's key
+// is never added to the contract. So he won't be able to add new messages. In
+// real life, we would only use the Bobs' public keys to configure the contract,
+// and only they would know their private keys.
+
+export const users = {
+  Bob: PrivateKey.fromBase58(
+    "EKFAdBGSSXrBbaCVqy4YjwWHoGEnsqYRQTqz227Eb5bzMx2bWu3F"
+  ),
+  SuperBob: PrivateKey.fromBase58(
+    "EKEitxmNYYMCyumtKr8xi1yPpY3Bq6RZTEQsozu2gGf44cNxowmg"
+  ),
+  MegaBob: PrivateKey.fromBase58(
+    "EKE9qUDcfqf6Gx9z6CNuuDYPe4XQQPzFBCfduck2X4PeFQJkhXtt"
+  ), // This one says duck in it :)
+  Jack: PrivateKey.fromBase58(
+    "EKFS9v8wxyrrEGfec4HXycCC2nH7xf79PtQorLXXsut9WUrav4Nw"
+  ),
+};
+
+export class Add extends SmartContract {
+  // On-chain variable definitions
```

Comment:
`// On-chain state definitions` would provide consistency with our idiomatic "on-chain state" phrase

Reply:
Out of curiosity, do some people use "variable" instead of "state" -- @jackryanservia?